src/scene/Scene.ts

```diff
@@ -81,6 +136,43 @@ class Scene {
     return this.frames;
   }
 
+  getSelectedElements(
+    opts: getSelectedElementsOpts,
+  ): NonDeleted<ExcalidrawElement>[] {
+    const hash = hashSelectionOpts(opts);
+
+    const elements = opts?.elements || this.nonDeletedElements;
+    if (
+      this.selectedElementsCache.elements === elements &&
+      this.selectedElementsCache.selectedElementIds === opts.selectedElementIds
+    ) {
+      const cached = this.selectedElementsCache.cache.get(hash);
+      if (cached) {
+        return cached;
+      }
+    } else if (opts?.elements == null) {
+      // if we're operating on latest scene elements and the cache is not
+      //  storing the latest elements, clear the cache
+      this.selectedElementsCache.cache.clear();
+    }
+
+    this.selectedElementsCache.selectedElementIds = opts.selectedElementIds;
```

Comment:
What is the purpose of setting the cache of selectedElementIds here, instead of doing it below?

Reply:
I should have moved it next to the other updates for sure, but the idea is to update it unconditionally, unlike the rest. Though that holds only if `opts.selectedElementIds` is guaranteed to come from the state as opposed to some adhoc selection (though I don't believe that would make sense)